### DISCUSSION TITLE
Problem bei Rechnung mit 2 Steuersätzen und generell

### DISCUSSION THREAD

**Discussion by SigiKoh** [stephanstapel/ZUGFeRD-csharp#459] [General]
ich muss als zugferd-neuling das xml implementieren und habe mich an das beispiel gehalten.
bei einer rechnung mit 2 steuersätzen werden im XML aber 3 einträge erzeugt, einer davon mit dem rechnungs-nettobetrag und steuer = 0.
verwendete mehtoden:
1 x invoice.SetTotals
2 x invoice.AddApplicableTradeTax
ist das korrekt? leider funktioniert der erzeugte XML-output mit keinem online-validator, dern ich finden konnte.
alle werfen eine lange liste an fehlern aus :-( wäre sehr dankbar für einen entsprechenden hinweis!

![grafik](https://github.com/user-attachments/assets/9a460ecd-fdf6-49f1-aaa1-e6fdaf3714a3)

**Comment by stephanstapel** [comment]
Ja, so ist es im Prinzip korrekt. Die dritte TradeTax muss in Deinem Code entstehen, schau mal, wo AddApplicableTradeTax() aufgerufen wird.

**Reply by SigiKoh** [reply]
oh gott, ich bin ein id..t - bitte vielmals um entschuldigung ... da war noch aus einem test eine dritte anweisung drin ... 
die habe ich jetzt 2 stunden lang nict gesehen :-( 
SORRY

**Reply by stephanstapel** [reply]
alles gut. Wenn Du Unterstützung bei der Implementierung brauchen solltest, lass' es mich wissen, ich biete gerne kommerziellen Support an, dann gerne über stephan(at)s2-industries.com
